src/Symfony/Component/DependencyInjection/Compiler/FactoryReturnTypePass.php

```diff
@@ -65,6 +66,9 @@ private function updateDefinition(ContainerBuilder $container, $id, Definition $
         if (is_string($factory)) {
             try {
                 $m = new \ReflectionFunction($factory);
+                if (false !== $m->getFileName() && file_exists($m->getFileName())) {
+                    $container->addResource(new FileResource($m->getFileName()));
```

Comment:
shouldn't we use your new reflection resource instead, to invalidate less often ?